include/lsst/afw/typehandling/Storable.h

```diff
@@ -101,8 +101,46 @@ class Storable : public table::io::Persistable {
      * cross-class comparisons are valid, implementers should take care that
      * they are symmetric and will give the same result no matter what the
      * compile-time types of the left- and right-hand sides are.
+     *
+     * @see singleClassEquals
      */
     virtual bool equals(Storable const& other) const noexcept;
+
+protected:
+    /**
+     * Test if a Storable is of a particular class and equal to another object.
+     *
+     * This method template simplifies implementations of @ref equals that
+     * delegate to `operator==` without supporting cross-class comparisons.
+     *
+     * @tparam T The class expected of the two objects to be compared.
+     * @param lhs, rhs The objects to compare. `lhs` is passed by pointer
+     *                 because it should almost always be `this` (see
+     *                 example below).
+     * @returns `true` if `rhs` is a `T` and `*lhs == rhs`; `false` otherwise.
+     *
+     * @exceptsafe Provides the same level of exception safety as `operator==`.
+     *             Most implementations of `operator==` do not throw.
+     *
+     * @note This method template calls `operator==` with arguments of type
+     *       `T const&`. Its use is *not* recommended if there would be any
+     *       ambiguity regarding which `operator==` gets called.
+     *
+     * This method template is typically called from @ref equals as:
+     *
+     *     bool MyType::equals(Storable const& other) const noexcept {
+     *         return singleClassEquals(this, other);
+     *     }
+     */
```

Comment:
This might be a context where providing a preprocessor macro would be useful, or perhaps a CRTP implementation mixin - something like:
```
template <typename Derived>
class EqualityComparableStorable : public Storable {
public:
    bool equals(Storable const & other) const noexcept override {
        Derived const & lhs = static_cast<Derived const &>(*this);
        ... singleClassEquals logic here...
    }
};
```
But of course macros have their own problems, and generalizing that CRTP idea to support more `Storable` methods is *at least* quite complicated (maybe one CRTP mixin, with bool template parameters to indicate which `Storable` operations to implement via various standard operators, and lots of partial specialization?).

Anyhow, what you have here is just fine and quite possibly better than either of those ideas; I'm just suggesting other options because you asked for them :smiley:.